lib/pbench/agent/tool_meister_start.py

```diff
@@ -296,8 +295,8 @@ def start_tms_via_ssh(
     cmd = f"{tool_meister_cmd} {redis_server.host} {redis_server.port} {{tm_param_key}} yes"
     if debug_level:
         cmd += f" {debug_level}"
-    template = TemplateSsh(ssh_cmd, shlex.split(ssh_opts), cmd)
-    tms = dict()
+    template = TemplateSsh(Path(ssh_cmd), shlex.split(ssh_opts), cmd)
```

Comment:
While it is true that the first argument to the `TemplateSsh()` constructor is typed as a `Path` [1], it really seems that it should be a string (`str`) instead.  If you agree, then this should not be changed, but instead we change `TemplateSsh` to accept a `str`.

[1] https://github.com/distributed-system-analysis/pbench/blob/main/lib/pbench/agent/utils.py#L400

Reply:
I guess you're right. I was sure I'd started that code depending on a `Path`; but looking at `main` it's not even a "path-like object" because I'm pulling it into the `argv` array without a `str(ssh_cmd)`. However, although the `subprocess.Popen` documentation doesn't seem to say anything to suggest this, it clearly does actually *work* as a path-like object, because it's being called both ways.

One however (from `start`) explicitly creates a `Path` from the `shutil.which`, while the other just passes a raw "ssh" string. It'd probably be better if both passed `shutil.which` output, and it'd probably be clearer to make it a string under the circumstances.

Reply:
I'm guessing that `subprocess.Popen()` wants a "command" for this argument, and, while that would typically be a _path_, it certainly doesn't have to be -- in appropriate circumstances I believe it could be a shell alias or a shell function.  So, if we want to be fully general, we should be requiring a string, not a `Path`.  (As for it "working"...Python's differentiation between `str` and `Path` is pretty thin, since they are both "path-like" and it seems to be trivial to convert between them...but, I'd like to see if we can do more than just "working" -- I would aspire toward making it "right".  😆)

Reply:
The thing is that `subprocess.Popen` takes an `argv` list; the first element in that list can apparently be "path-like" while the others are just strings. There must be code somewhere that's treating the first element specially... or maybe it's all just buried down in C code with some interface magic. Weird, in any case, and I don't particularly like that this happens transparently ... or that I managed to get the type hint wrong but it doesn't matter. 😵‍💫